ansi/width.go

```diff
@@ -79,16 +78,16 @@ func StringWidth(s string) int {
 
 	for i := 0; i < len(s); i++ {
 		state, action := parser.Table.Transition(pstate, s[i])
+		if state == parser.Utf8State {
+			var w int
+			cluster, _, w, gstate = uniseg.FirstGraphemeClusterInString(s[i:], gstate)
+			width += w
+			i += len(cluster) - 1
+			pstate = parser.GroundState
+			continue
+		}
 		switch action {
 		case parser.PrintAction:
-			if utf8ByteLen(s[i]) > 1 {
-				var w int
-				cluster, _, w, gstate = uniseg.FirstGraphemeClusterInString(s[i:], gstate)
-				width += w
-				i += len(cluster) - 1
-				pstate = parser.GroundState
-				continue
-			}
 			width++
 			fallthrough
```

Comment:
This padas through is over kill as the code in the default can be moved out after the switch, and fall through and default could be removed